firmware/controllers/system/ThreadController.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "ControllerBase.h"
+
+#include "ch.h"
+
+template <int TStackSize>
```

Comment:
@rusefi Explanation of why this is templated: Because we want to statically allocate the thread's stack (but different controllers might need different size stacks), we have to let the compiler know how big it should be.

Reply:
this part I like. also if you feel to comment on something in PR this a sign that you need a code comment :)

Reply:
yep, writing in-source docs are in the cards

Reply:
docs added